src/server/stats/stats_mixin.js

```diff
@@ -0,0 +1,19 @@
+import { getStats } from './stats';
+
+export function statsMixin(kbnServer, server) {
+  server.route({
+    method: 'GET',
+    path: '/api/stats',
+    handler: async function (request, reply) {
+      const { callWithInternalUser } = server.plugins.elasticsearch.getCluster('admin');
+      const callAdminCluster = (...args) => callWithInternalUser(...args);
+
+      const stats = await getStats(
+        server.config().get('kibana.index'),
+        callAdminCluster
+      );
+
+      return reply(stats);
```

Comment:
`async` / `await` is all just sugar on top of `Promise`, so the method is already returning a `Promise` that you choose to `await` against to get the value (or throw the exception). So I'm guessing if it worked, then you simply can stick with what you've got and can avoid my extra fluff.